### ISSUE TITLE
Camera Artifacs

### ISSUE THREAD

**Issue by RodrigoMedeirosrRS** [tobspr/RenderPipeline#67]
Right, i found a little glitch when i try to resposition my câmera.

Here is a screenshot before i active the function:
![erro01](https://cloud.githubusercontent.com/assets/7956669/21526788/07fe81c8-cd27-11e6-8238-119bd961465f.png)
... and here is a screenshot after i active the function:
![erro02](https://cloud.githubusercontent.com/assets/7956669/21526798/1cb9c2a8-cd27-11e6-9bf1-97faf8fef80e.png)

And here is the function:

def CameraReposition(self):
       base.camera.reparentTo(self.point)
       base.camera.lookAt(self.orb)

The problem is directly related with the "reparentTo" command.

**Comment by tobspr** [comment]
Hi, yeah,  the render pipeline does not expect the camera to be parented to another nodepath. 

I will see if I get to implement this for the next Release, in the meantime you can use a task which synchronises the camera position to the desired nodepath.

**Comment by RodrigoMedeirosrRS** [comment]
Right! I will wait for you!

Please try to fix it into next release, because into my controller system, when i update the camera position into task, the results very, very poor.

Thanks for the attetion.

**Comment by tobspr** [comment]
Make sure the task has the right sort, so that your logic happens first, then the camera position is synchronized, and the render pipeline updates. 

The rp updares all have a sort greater zero, so try running your game logic with a sort of -10 and your camera synchronization with a sort of -5 or so. 

I am currently in vacation, so I can't do much until next week :)

**Comment by RodrigoMedeirosrRS** [comment]
Right! Thanks man!

**Comment by RodrigoMedeirosrRS** [comment]
I found the origin of problem!

Analyzing the glitchs, i see they follow a pattern, and they occour's only into skybox, and how many times i test, less glichs occour's. So when i try you manual solution suggestion, i find the origin of error, and a perfect workaround.

The error occour's because the RP cannot get the starter position of camera when his is a child of another thing, and he is getting some memory info trash, and receiving invalid coordinates.

But, if i can set manualy only the original position, the RP can apply any other transformations correctly, and here is the workaround:

base.camera.setPos(self.orb.getPos() + self.playerNP.getPos() + Vec3(0, 2.5, 0))
base.camera.lookAt(self.orb)
base.camera.setPos(0,2.5,0)
base.camera.lookAt(self.orb)

Basically, i am manualy setting the camera position into hes future child position, then i make it a child, after this, the RP can correctly calculate the camera position and all glich's are solved.

Now i belive is much more easy to you to found a definitive solution ;)

Att. R.M.Lehnemann

**Comment by RodrigoMedeirosrRS** [comment]
No more fix is needed, problem as definitely solved.

If you need reparent a camera, instead create a empty node, them use the following command:

base.camera.setTransform(self.EmptyNode.getNetTransform())

Solved. The camera now will recive all transforms of child component.

**Comment by tobspr** [comment]
Cool, glad to hear you found a solution!

I will leave this PR open though, since I think the render pipeline should automatically be able to detect this (by always using the correct transform)
